src/app/Fake.Core.Trace/Trace.fs

```diff
@@ -232,11 +255,33 @@ let traceEndTaskUnsafe task = traceEndTaskUnsafeEx TagStatus.Success task
 /// Traces the end of a task
 [<System.Obsolete("Consider using traceTask instead and 'use' to properly call traceEndTask in case of exceptions. To remove this warning use 'traceEndTask'.")>]
 let traceEndTask task = traceEndTaskUnsafe task
-     
+
+/// Wrap functions in a 'use' of this function     
 let traceTask name description =
     traceStartTaskUnsafe name description
     asSafeDisposable (fun state -> traceEndTaskUnsafeEx state name)
 
+/// Traces a function execution
+/// If no exception is thrown then trace is marked as success
+/// Any exception thrown will result in a mark failed and exception re-thrown 
+let traceFunction name description func =
```

Comment:
What about an in-line high order function
```fsharp
useWith (traceTask name description) (fun _ ->
   // callback
)
```

implemented like

```fsharp
let inline useWith (d:ISafeDisposable) f =
  use t = d
  try let result = f () in t.MarkSuccess(); result with _ -> t.MarkFailed(); reraise()
```

Reply:
i like it, implemented :)